commit 79a42de
Author: Chad A Davis <chad.davis@booking.com>
Date:   2012-01-21 16:45:37 +0100

    Patch git-completion to read config option bash.onlyHEADs

diff --git a/git-completion.sh b/git-completion.sh
index a7d20df..ba68e84
--- a/git-completion.sh
+++ b/git-completion.sh
@@ -550,6 +550,10 @@ __git_tags ()
 # by checkout for tracking branches
 __git_refs ()
 {
+ if [ "$(git config --bool bash.onlyHEADs)" == "true" ]; then
+        __git_heads
+        return
+    fi
        local i is_hash=y dir="$(__gitdir "${1-}")" track="${2-}"
        local cur format refs
        _get_comp_words_by_ref -n =: cur
